compiler/src/iree/compiler/Codegen/Common/GPU/GPUPromoteMatmulOperands.cpp

```diff
@@ -186,32 +139,51 @@ struct GPUPromoteMatmulOperandsPass final
     FunctionOpInterface funcOp = getOperation();
 
     OpBuilder builder(funcOp);
-    funcOp.walk([&](Operation *op) {
+    WalkResult walkResult = funcOp.walk([&](Operation *op) {
       auto loweringConfig =
           getLoweringConfig<IREE::GPU::LoweringConfigAttr>(op);
       if (!loweringConfig) {
-        return;
+        return WalkResult::advance();
       }
 
       std::optional<SmallVector<int64_t>> promotedOperands =
           getPromotedOperandList(loweringConfig);
       if (!promotedOperands) {
-        return;
+        return WalkResult::advance();
       }
 
-      SmallVector<bool> useDirectLoad =
-          getUseDirectLoad(loweringConfig)
-              .value_or(SmallVector<bool>(promotedOperands->size(), false));
+      std::optional<ArrayRef<Attribute>> maybePromotionTypes =
+          getPromotionTypesList(loweringConfig);
+      if (maybePromotionTypes && maybePromotionTypes.value().size() !=
+                                     promotedOperands.value().size()) {
```

Comment:
nit/optional if you want to be more concise:

```suggestion
      if (maybePromotionTypes && maybePromotionTypes->size() !=
                                     promotedOperands->size()) {
```